pkg/pillar/types/edgeviewtypes.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2021-2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Types which feed in and out of the verifier
+
+package types
+
+import "time"
+
+// Types for edge-view on device
+
+const (
+	// EdgeviewPath - path to edgeview files
+	EdgeviewPath = "/run/edgeview/"
+	// EdgeviewJwtFile - location for JWT string
+	EdgeviewJwtFile = EdgeviewPath + "edge-view-token"
+	// EdgeviewExpFile - expiration file
+	EdgeviewExpFile = EdgeviewPath + "edge-view-notafter"
+	// EdgeviewKeyFile - authentication nonce file
+	EdgeviewKeyFile = EdgeviewPath + "edge-view-nonce"
+	// EdgeviewCertFile - dispatcher cert pem file
+	EdgeviewCertFile = "/persist/certs/wss-server-cacert.pem"
+	// EdgeviewPolicyDev - dev policy json file
+	EdgeviewPolicyDev = EdgeviewPath + "devPolicy.json"
+	// EdgeviewPolicyApp - app policy json file
+	EdgeviewPolicyApp = EdgeviewPath + "appPolicy.json"
+)
+
+// EdgeviewConfig - edge-view config from controller
+type EdgeviewConfig struct {
+	JWToken     string      // JWT token for edge-view
+	DispCertPEM [][]byte    // dispatcher certificates
+	DevPolicy   EvDevPolicy // device policy
+	AppPolicy   EvAppPolicy // app policy
+}
+
+// EvDevPolicy - edge-view policy for device access
+// the 'Enabled' controls device side is allowed or not including debug commands
+// With Enable Dev, can expend later for other policies
+type EvDevPolicy struct {
+	Enabled bool `json:"enabled"` // allow access to device
+}
+
+// EvAppPolicy - edge-view policy for application access
+// the 'Enabled' controls all app access is allowed or not
+// With Enable App, can expend later for other policies
+type EvAppPolicy struct {
+	Enabled bool `json:"enabled"` // allow access to apps
+}
+
+// EvjwtAlgo - jwt algorithm
+// JWT token for edgeview
+// JWT has 3 portion of items separated by '.' using base64url without padding,
+// the 1st part is the algorithm, the 2nd is the info, the third is signing data
+// the 1st and 2nd parts are from json format
+type EvjwtAlgo struct {
+	Alg string `json:"alg"` // algorithm, use 'ES256' or SHA256withECDSA
+	Typ string `json:"typ"` // type, is 'JWT' string
+}
+
+// EvjwtInfo - token embedded info
+// the info specifies where is the dispatcher endpoint, the intended EVE
+// device with UUID string, the token expiration time and authentication nonce
+type EvjwtInfo struct {
+	Dep string `json:"dep"` // dispatcher end-point string e.g. ip:port
+	Sub string `json:"sub"` // jwt subject, the device UUID string
+	Exp int    `json:"exp"` // expiration time for the token
+	Key string `json:"key"` // key or nonce for payload hmac authentication
```

Comment:
Rename field to nonce?

Reply:
this is 3 letter normally, may be Nnc ?

Reply:
Does some specification require it to be exactly three characters? If so keep it the way it is.

Reply:
OK